Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/storage/retention/CheckInCleaner.kt

```diff
@@ -0,0 +1,34 @@
+package de.rki.coronawarnapp.eventregistration.storage.retention
+
+import dagger.Reusable
+import de.rki.coronawarnapp.eventregistration.checkins.CheckInRepository
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.seconds
+import de.rki.coronawarnapp.util.TimeStamper
+import kotlinx.coroutines.flow.first
+import org.joda.time.DateTimeConstants
+import org.joda.time.Instant
+import javax.inject.Inject
+
+@Reusable
+class CheckInCleaner @Inject constructor(
+    private val checkInRepository: CheckInRepository,
+    private val timeStamper: TimeStamper
+) {
+
+    suspend fun cleanUp() {
+        val checkInsToDelete = checkInRepository.allCheckIns.first()
+            .filter {
+                isOutOfRetention(it.checkInEnd)
+            }
+        checkInRepository.deleteCheckIns(checkInsToDelete)
+    }
+
+    private fun isOutOfRetention(checkInEndDate: Instant): Boolean {
```

Comment:
Thanks. I now calculate it once per run. Delete per SQL statement would be most efficient I guess, however it is also harder to implement and test, and since there are not that many entries in these tables I think its okay as it is now.